util/mp.py

```diff
@@ -1,10 +1,51 @@
 from __future__ import absolute_import, division, print_function
 
+import os
+
 import future.moves.itertools as itertools
+import psutil
 
 import libtbx.easy_mp
 
 
+def available_nproc():
+    """
+    Choose a suitable value for nproc based on the availability of processors.
+    """
```

Comment:
```suggestion
def available_cores() -> int:
    """
    Determine the number of available processor cores.
    There are a number of different methods to get this information, some of which may not
    be available on a specific OS and/or version of Python. So try them in order and return
    the first successful one.
    """
```

Reply:
can do - took `nproc` as being the accepted jargon in DIALS, but `cores` seems better. Given your improved docstring I'll remove the comments I added